Assets/Scripts/Construction/PlayerConstructionHandler.cs

```diff
@@ -0,0 +1,157 @@
+        [SerializeField] ConstructionPreview constructionPreview;
+        [SerializeField] LayerMask buildTargetingCollisionMask;
+        [SerializeField] float maxBuildDistance = 20.0f;
+        [SerializeField] float maxTerrainHeight = 100.0f;
+        [SerializeField] GameObject constructionSitePrefab;
+        [SerializeField] BuildingInfo initialBuilding; // TEMP
+
+        private DefaultInputActions actions;
+        private Vector3? currentPlacementPosition;
+        private BuildingInfo currentBuildingInfo;
+        private int currentConstructionRotationSteps = 0; // multiply by 90deg to get actual rotation
+        private Quaternion CurrentConstructionRotation => Quaternion.Euler(0, currentConstructionRotationSteps * 90.0f, 0);
+        private bool isPlacementActive = false;
+
+        void LateUpdate()
+        {
+            ProcessPlacement();
+        } 
+
+        // Select a building and start scanning for placement position
+        // TODO: hook this up to a state machine or something so it works with the rest of the player mechanics
+        public void InitiatePlacement(BuildingInfo buildingInfo) 
+        {
+            currentBuildingInfo = buildingInfo;
+            isPlacementActive = true;
+            constructionPreview.SetBuilding(buildingInfo);
+            constructionPreview.gameObject.SetActive(true);
+        }
+        
+        // Place the construction plot
+        private void ExecutePlacement() 
+        {
+            if (isPlacementActive == false){
+                Debug.LogWarning("Attempted to place building outside of placement mode!");
+                return;
+            }
+            if (currentPlacementPosition is null) {
+                Debug.LogWarning("Attempted to place building in impossible position");
+                ExitPlacement();
+                return;
+            }
+
+            GameObject newSite = Instantiate(constructionSitePrefab, (Vector3)currentPlacementPosition, CurrentConstructionRotation);
+            newSite.GetComponent<ConstructionSite>().Initialize(currentBuildingInfo);
+
+            ExitPlacement();
+        }
+
+        // Exit placement mode
+        public void ExitPlacement() 
+        {
+            currentBuildingInfo = null;
+            isPlacementActive = false;
+            constructionPreview.gameObject.SetActive(false);
+        }
+
+        private void ProcessPlacement() 
+        {
+            if (isPlacementActive) {
+                Vector2Int effectiveGridSize = currentBuildingInfo.GetEffectiveGridSize(currentConstructionRotationSteps);
+                bool centerSnapX = effectiveGridSize.x % 2 == 1;
+                bool centerSnapZ = effectiveGridSize.y % 2 == 1;
+                Vector3 snappedBuildPoint = SnapToGrid(GetBuildPoint(), centerSnapX, centerSnapZ);
+                currentPlacementPosition = ScanBuildingPlacement(snappedBuildPoint, currentBuildingInfo.GridSize, CurrentConstructionRotation);
+            }
```

Comment:
It's always better to invert the statements to avoid needless nesting. In this case:
```if (isPlacementActive is false) return;```